src/clr-angular/utils/drag-and-drop/providers/drag-and-drop-dispatcher.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+import {DraggableEvent} from "../interfaces/draggable-event";
+
+@Injectable()
+export class DragAndDropDispatcher {
+    private _onDragStart: Subject<DraggableEvent> = new Subject<DraggableEvent>();
+    private _onDragMove: Subject<DraggableEvent> = new Subject<DraggableEvent>();
+    private _onDragEnd: Subject<DraggableEvent> = new Subject<DraggableEvent>();
+    private _onDrop: Subject<void> = new Subject<void>();
+
+    get onDragStart(): Observable<DraggableEvent> {
+        return this._onDragStart.asObservable();
+    }
+
+    get onDragMove(): Observable<DraggableEvent> {
+        return this._onDragMove.asObservable();
+    }
+
+    get onDragEnd(): Observable<DraggableEvent> {
+        return this._onDragEnd.asObservable();
+    }
+
+    get onDrop(): Observable<void> {
+        return this._onDrop.asObservable();
+    }
+
+    dragStart(dragEvent: DraggableEvent) {
```

Comment:
nit - @youdz do we have a naming convention for these types of methods where you are just calling the `next` method on the `Subject`. I think naming this method as `notifyDragStart` instead of just `dragStart` would help better understand what it is doing.

Reply:
I don't think a convention brings much here, other than unnecessary verbosity. To be fair, I think a more natural method name would be `startDrag()`, but anything including `notify` feel overkill.